lib/Optimizer/Optimization.h

```diff
@@ -0,0 +1,9 @@
+#include "glow/Graph/Graph.h"
+
+class Optimization {
+public:
+  Optimization() {}
+  virtual ~Optimization(){};
+  virtual bool run(glow::Function *F) { return false; }
```

Comment:
This feels like a good candidate for "interface". I'd make `run` and `getName` pure virtual.